test/jdk/java/nio/channels/AsynchronousFileChannel/Basic.java

```diff
@@ -563,15 +565,22 @@ static void testTruncate(Path file) throws IOException {
     static ByteBuffer genBuffer() {
         int size = 1024 + rand.nextInt(16000);
         byte[] buf = new byte[size];
-        boolean useDirect = rand.nextBoolean();
-        if (useDirect) {
-            ByteBuffer bb = ByteBuffer.allocateDirect(buf.length);
-            bb.put(buf);
-            bb.flip();
-            return bb;
-        } else {
-            return ByteBuffer.wrap(buf);
-        }
+        return switch (rand.nextInt(3)) {
+            case 0 -> {
+                ByteBuffer bb = ByteBuffer.allocateDirect(buf.length);
+                bb.put(buf);
+                bb.flip();
+                yield bb;
+            }
+            case 1 -> ByteBuffer.wrap(buf);
+            case 2 -> {
+                ByteBuffer bb = Arena.ofAuto().allocate(buf.length).asByteBuffer();
+                bb.put(buf);
+                bb.flip();
+                yield bb;
+            }
+            default -> throw new InternalError("Should not reach here");
+        };
```

Comment:
```suggestion
        return switch (rand.nextInt(3)) {
            case 0  -> ByteBuffer.allocateDirect(buf.length)
                                 .put(buf)
                                 .flip();
            case 1  -> ByteBuffer.wrap(buf);
            case 2  -> Arena.ofAuto()
                            .allocate(buf.length)
                            .asByteBuffer()
                            .put(buf)
                            .flip();
            default -> throw new InternalError("Should not reach here");
        };
```
ByteBuffer supports chain programming style, so we can simplify it to this